### ISSUE TITLE
Composed linear layers?

### ISSUE THREAD

**Issue by apeguero1** [tatp22/linformer-pytorch#6]
Hey @tatp22 great repo! 

I'm having trouble wrapping my head around the `w_q, w_k, and w_v ` linear layers in the `LinearAttentionHead` module. Are they needed? There's no activation between the previous linear layers, `to_q, to_k, to_v` in `MHAttention`, and those weights so they wouldn't add any expressivity to the model since you would just be multiplying two matrices together which is equivalent to one linear layer. The `E` and `F` projections also seem like they're being composed with `w_k, and w_v ` without a non-linearity.

Looking at Eq. 7 from the [paper](https://arxiv.org/pdf/2006.04768.pdf) your implementation seems correct though.

Any thoughts on this?

**Comment by tatp22** [comment]
Hi @apeguero1, thank you for pointing this out!

It seems as though you are correct. Looking at the code, It looks like I don't need the `w_q`, `w_k`, and `w_v` matrices, if I make sure that for each head, there is a `to_q`, `to_k`, and `to_v` matrix that is an `nn.Linear(channels, dim)` layer. This would mean that instead of having only one `to_q`, `to_k`, and `to_v` matrix for each `MHAttention`, there would be `nhead` of these matrices. This would mean that, essentially, the `LinearAttentionHead` module would just become a matrix multiplication with no learnable parameters, besides the `E` and `F` layers.

I will make a pull request later of this change, let me know what you think!

**Comment by apeguero1** [comment]
Great sounds good, I'll take a look!

**Comment by tatp22** [comment]
Check it out and see what you think, let me know if you see any errors :+1:

~~The only problem is that training takes longer, i don't know if that's just a thing with my computer and it not running so well, so if you see something that could affect that, please let me know!~~ <- Nevermind that

**Comment by apeguero1** [comment]
Yep looks good to me. Thanks for the quick response! (:

**Comment by tatp22** [comment]
No prob :) Merged, and the latest version is available with version `0.10.0` on `pip`
